cs.html

```diff
@@ -313,6 +322,146 @@ <h2>Methods</h2>
+                <li>
+                  <p>If <code><var>data</var></code> is too large to fit into a
+                  datagram, return a promise <a>rejected</a> with a newly created
+                  <code>InvalidArgumentError</code> and abort these steps.</p>
+                </li>
+                <li>
+                  <p>If <var>transport</var> is unable to send the datagram due
+                  to congestion control, return a promise <a>rejected</a> with
+                  a newly created <code>InvalidStateError</code> and abort
+                  these steps.</p>
+                </li>
+                <li>
+                  <p>Let <var>p</var> be a new promise.</p>
+                </li>
+                <li>
+                  <p>Return <var>p</var> and continue the following steps in
+                  the background.</p>
+                </li>
+                <ol>
+                  <li>
+                    <p>Send <var>data</var> in a QUIC datagram.</p>
+                  </li>
+                  <li>
+                    <p>When an ack is received for the sent datagram,
+                    <a>resolve</a> <var>p</var> with <code>true</code>.</p>
+                  </li>
+                  <li>
+                    <p>When the datagram is detemined to be lost, <a>resolve</a>
+                    <var>p</var> with <code>false</code>.</p>
+                  </li>
+                </ol>
+              </ol>
+              <table class="parameters">
+               <tbody>
+                  <tr>
+                    <th>Parameter</th>
+                    <th>Type</th>
+                    <th>Nullable</th>
+                    <th>Optional</th>
+                    <th>Description</th>
+                  </tr>
+                  <tr>
+                    <td class="prmName">data</td>
+                    <td class="prmType"><code>Uint8Array</code></td>
+                    <td class="prmNullFalse"><span role="img" aria-label=
+                    "False">&#10008;</span></td>
+                    <td class="prmOptFalse"><span role="img" aria-label=
+                    "False">&#10008;</span></td>
+                    <td class="prmDesc"></td>
+                  </tr>
+                </tbody>
+              </table>
+              <div>
+                <em>Return type:</em> <code>Promise&lt;boolean&gt;</code>
+              </div>
+            </dd>
+            <dt><dfn><code>receiveDatagrams</code></dfn></dt>
+            <dd>
+              <p>If datagrams have been received since the last call to receiveDatagrams(),
+		 return a new promise resolved with all of the received datagrams. </p>
+              <p>If not, return a new promise that will resolve when more datagrams are received, 
+		 resolved with all datagrams received. </p>
+              <p>If too many datagrams are queued between calls to receiveDatagrams(), 
+                 the implementation may drop datagrams and replace them with a null value
```

Comment:
Will null values always be at the end of the sequence in this case?

Reply:
No, it could put them anywhere